internal/supervisor/cluster.go

```diff
@@ -33,8 +33,9 @@ type ClusterProvisioner interface {
 	PrepareCluster(cluster *model.Cluster) bool
 	CreateCluster(cluster *model.Cluster) error
 	CheckClusterCreated(cluster *model.Cluster) (bool, error)
-	CreateNodes(cluster *model.Cluster) error
-	CheckNodesCreated(cluster *model.Cluster) (bool, error)
+	CreateNodegroups(cluster *model.Cluster) error
+	CheckNodegroupsCreated(cluster *model.Cluster) (bool, error)
+	DeleteNodegroup(cluster *model.Cluster) error
```

Comment:
My only nitpick here is that if this can remove more than one nodegroup it should be plural, even if the CLI only allows for one at a time?

Reply:
Right. Thanks